src/Microsoft.VisualStudio.Web.CodeGeneration.ProjectInfo/MsBuildProjectFile.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.ProjectInfo
+{ 
+    public class MsBuildProjectFile
+    {
+        private readonly IEnumerable<string> _sourceFiles;
+        private readonly IEnumerable<string> _projectReferences;
+        private readonly IEnumerable<string> _assemblyReferences;
+        private readonly IDictionary<string, string> _properties;
+        private readonly string _fullPath;
+
+        public MsBuildProjectFile(string fullPath,
+            IEnumerable<string> sourceFiles,
+            IEnumerable<string> projectReferences,
+            IEnumerable<string> assemblyReferences,
+            IDictionary<string, string> properties)
+        {
+            Requires.NotNullOrEmpty(fullPath);
+            Requires.NotNull(sourceFiles);
+            Requires.NotNull(projectReferences);
+            Requires.NotNull(assemblyReferences);
+            Requires.NotNull(properties);
+
+            _sourceFiles = sourceFiles;
+            _projectReferences = projectReferences;
+            _assemblyReferences = assemblyReferences;
+            _properties = properties;
+            _fullPath = fullPath;
+        }
+
+        public IEnumerable<string> SourceFiles
```

Comment:
The getters aren't really useful. Can simplify by using "private set" properties instead.